extensions/notebook/src/jupyter/jupyterController.ts

```diff
@@ -196,16 +202,31 @@ export class JupyterController implements vscode.Disposable {
 		});
 	}
 
-	public doManagePackages(): void {
+	public async doManagePackages(): Promise<void> {
 		try {
-			let packagesDialog = new ManagePackagesDialog(this._jupyterInstallation);
+			let model = new ManagePackageDialogModel(this._jupyterInstallation, this._packageManageProviders);
+			await model.init();
+			let packagesDialog = new ManagePackagesDialog(model);
 			packagesDialog.showDialog();
 		} catch (error) {
 			let message = utils.getErrorMessage(error);
 			this.apiWrapper.showErrorMessage(message);
 		}
 	}
 
+	public registerPackageManagers(providerId: string, packageManageProvider: IPackageManageProvider): void {
+		if (packageManageProvider) {
+			if (!this._packageManageProviders.has(providerId)) {
+				this._packageManageProviders.set(providerId, packageManageProvider);
```

Comment:
>_packageManageProviders [](start = 9, length = 23)

Seems like it should be an error case (at least log something) if a duplicate provider is registered